platform/darwin/src/MGLStyle.mm

```diff
@@ -92,19 +92,11 @@ - (NSString *)name {
     return @(self.mapView.mbglMap->getStyleName().c_str());
 }
 
-- (mbgl::style::Layer *)mbglLayerWithIdentifier:(NSString *)identifier
-{
-    return self.mapView.mbglMap->getLayer(identifier.UTF8String);
-}
-
-- (mbgl::style::Source *)mbglSourceWithIdentifier:(NSString *)identifier
-{
-    return self.mapView.mbglMap->getSource(identifier.UTF8String);
-}
-
 - (id <MGLStyleLayer>)layerWithIdentifier:(NSString *)identifier
 {
     auto layer = self.mapView.mbglMap->getLayer(identifier.UTF8String);
+
+    if (layer == nullptr) return nil;
```

Comment:
(By the way, this can be expressed more succinctly as `if (layer) return nil;`.)